src/pages/ImportPage.tsx

```diff
@@ -0,0 +1,756 @@
+import { useEffect, useState } from "react";
+import {
+  bulkRemove,
+  bulkUpsert,
+  clear,
+  getData,
+  getAllData,
+  remove,
+  upsert,
+} from "../indexedDB";
+import {
+  Button,
+  ComposedModal,
+  ContentSwitcher,
+  ContentSwitcherOnChangeData,
+  DataTable,
+  DataTableHeader,
+  DataTableRow,
+  DenormalizedRow,
+  FileUploader,
+  Loading,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  OverflowMenu,
+  OverflowMenuItem,
+  Pagination,
+  Switch,
+  Table,
+  TableBatchAction,
+  TableBatchActions,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableSelectRow,
+  TableToolbar,
+  TableToolbarAction,
+  TableToolbarContent,
+  TableToolbarMenu,
+  TableToolbarSearch,
+  TextInput,
+  ToastNotification,
+} from "carbon-components-react";
+import Papa, { ParseResult } from "papaparse";
+import { Edit, TrashCan } from "@carbon/icons-react";
+import axios from "axios";
+
+const ImportPage = () => {
+  type ModalType = "import" | "add" | "edit" | "delete" | "editHeader" | "";
+
+  const [rows, setRows] = useState<Array<DataTableRow>>([]);
+  const [headers, setHeaders] = useState<Array<DataTableHeader>>([]);
+  const [page, setPage] = useState<number>(1);
+  const [pageSize, setPageSize] = useState<number>(10);
+  const [open, setOpen] = useState<ModalType>("");
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [targetId, setTargetId] = useState<number | number[]>(NaN);
+  const [editData, setEditData] = useState<any>({});
+  const [showNotification, setShowNotification] = useState<
+    "success" | "error" | ""
+  >("");
```

Comment:
If we have decided to go with React Query and the loaders feature, the rest of the logic can be written here using useReducer which help us handle such cases more efficiently